### PR TITLE
Allow Configurable Processing times.

### PR DESCRIPTION
**Pull request by Leigham**
<Please review the guidelines for contributing to this repository. The link is above.>

Resolves #<issue ID here>. <If applicable.>

#### Changes proposed in this pull request: 
-  Changing process times based on amount of items being processed, and a time multiplier to allow people to put different processing times for each item.

### CODE REVIEW

File: Altis_Life.Altis/core/actions/fn_processAction.sqf [hunk 1]
```diff
@@ -8,7 +8,7 @@
     Master handling for processing an item.
     NiiRoZz : Added multiprocess
 */
-private ["_vendor","_type","_itemInfo","_oldItem","_newItemWeight","_newItem","_oldItemWeight","_cost","_upp","_hasLicense","_itemName","_oldVal","_ui","_progress","_pgText","_cP","_materialsRequired","_materialsGiven","_noLicenseCost","_text","_filter","_totalConversions","_minimumConversions"];
+private ["_processMultiplier","_vendor","_type","_itemInfo","_oldItem","_newItemWeight","_newItem","_oldItemWeight","_cost","_upp","_hasLicense","_itemName","_oldVal","_ui","_progress","_pgText","_cP","_materialsRequired","_materialsGiven","_noLicenseCost","_text","_filter","_totalConversions","_minimumConversions"];
```
Comment (Jason2605) [inline Altis_Life.Altis/core/actions/fn_processAction.sqf]: Maybe we can change this so the privates are infant of the variable declarations?

File: Altis_Life.Altis/core/actions/fn_processAction.sqf [hunk 2]
```diff
@@ -90,18 +91,25 @@ _pgText = _ui displayCtrl 38202;
 _pgText ctrlSetText format ["%2 (1%1)...","%",_upp];
 _progress progressSetPosition 0.01;
 _cP = 0.01;
-
 life_is_processing = true;
 
+//Making the process per item, configurable.
+_baseTime = 0.01;
+_sleepTime = ((_processMultiplier * _minimumConversions) * _baseTime);
```
Comment (Jason2605) [inline Altis_Life.Altis/core/actions/fn_processAction.sqf]: No need for the brackets really.

File: Altis_Life.Altis/core/actions/fn_processAction.sqf [hunk 3]
```diff
@@ -90,18 +91,25 @@ _pgText = _ui displayCtrl 38202;
...
-        uiSleep  0.28;
-        _cP = _cP + 0.01;
-        _progress progressSetPosition _cP;
-        _pgText ctrlSetText format ["%3 (%1%2)...",round(_cP * 100),"%",_upp];
-        if (_cP >= 1) exitWith {};
-        if (player distance _vendor > 10) exitWith {};
-    };
+  _timeBefore = diag_tickTime;
+  for "_i" from 0 to 1 step 0 do {
+       uiSleep  _sleepTime;
```
Comment (Jason2605) [inline Altis_Life.Altis/core/actions/fn_processAction.sqf]: Two spaces.

File: Altis_Life.Altis/core/actions/fn_processAction.sqf [hunk 4]
```diff
@@ -90,18 +91,25 @@ _pgText = _ui displayCtrl 38202;
...
+       _cP = _cP + 0.01;
+       _progress progressSetPosition _cP;
+       _pgText ctrlSetText format ["%3 (%1%2)...",round(_cP * 100),"%",_upp];
+       if (_cP >= 1) exitWith {
+         _timeAfter = diag_tickTime;
```
Comment (Jason2605) [inline Altis_Life.Altis/core/actions/fn_processAction.sqf]: Does this really need to be a variable?

File: Altis_Life.Altis/core/actions/fn_processAction.sqf (hunk 3 above)
Comment (SimZor) [reply Altis_Life.Altis/core/actions/fn_processAction.sqf]: Two spaces to much

File: Altis_Life.Altis/core/actions/fn_processAction.sqf (hunk 2 above)
Comment (SimZor) [reply Altis_Life.Altis/core/actions/fn_processAction.sqf]: Parenthesis*

Comment (Oly396) [reply Altis_Life.Altis/core/actions/fn_processAction.sqf]: Parentheses*

**Comment by Leigham** [comment]
just realised, I dont think what was put in there for checking itemWeight, doesnt work for multiItem processing.

File: Altis_Life.Altis/core/actions/fn_processAction.sqf (hunk 3 above)
Comment (Oly396) [reply Altis_Life.Altis/core/actions/fn_processAction.sqf]: Two space too much*

**Comment by DeliciousJaffa** [comment]
Should this not be implemented as actual time rather than a multiplier, actual amount of time would be much clearer to define and understand while producing the same result.

**Comment by Leigham** [comment]
@DeliciousJaffa  Im working on that, its not perfect, 1 second in the config * 1 item = 1.6 seconds. 10 items = 10.2 seconds , so the more items, the closer the math, but im redoing the method right now.

**Comment by Jason2605** [comment]
@Leigham If you are redoing the method does this PR still apply?

**Comment by RosieCode95** [comment]
closing due to 5 weeks of inactivity and outdated branch, @Leigham please reopen if you want to finish this

**Pull request closed without merging**